### DISCUSSION TITLE
Ho do you set the terrain so the sculpting tools to work? The video tutorial its assuming this just work but it doesn't for me.

### DISCUSSION THREAD

**Discussion by dilectiogames** [TokisanGames/Terrain3D#496] [Questions] [answered]
I've downloaded the latest version of the plugin 0.9.2 in godot 4.3

opening the demo all works and i can sculpt and paint as expected.

I've added the addon in my project and I've created a terrain. but it doesn't work. sculpting and painting tools appear to do nothing when trying to paint. then if deselect the terrain and select it again while it remains flat with nothing showing if you enable collisions view it shows that the terrain is deformed.

what step of the setup i'm missing?

**Comment by dilectiogames** [comment]
this plugin doesn't work on compatibility mode? switching the project to forward works correctly

**Reply by TokisanGames** [reply, marked as answer]
Compatibility mode is not yet supported. Work has been done on it. It will probably have limited supported in 0.9.4.

https://terrain3d.readthedocs.io/en/latest/docs/installation.html#requirements
https://terrain3d.readthedocs.io/en/latest/docs/mobile_web.html#webgl

**Reply by dilectiogames** [reply]
thank you, struggling with this for a few hours not understanding why it didn't wanted to work
